tests/models/test_ngsi_ld_entities.py

```diff
@@ -0,0 +1,533 @@
+        self.cb_client.post_entity(entity=self.entity)
+        ret_entity = self.cb_client.get_entity(entity_id=self.entity.id)
+        self.assertEqual(ret_entity.id,self.entity.id)
+        self.assertEqual(ret_entity.type,self.entity.type)
+
+        """Test2""" 
+        with self.assertRaises(requests.exceptions.HTTPError) as contextmanager:
+            self.cb_client.get_entity("urn:roomDoesnotExist")
+        response = contextmanager.exception.response
+        self.assertEqual(response.status_code, 404)
+
+        with self.assertRaises(requests.exceptions.HTTPError) as contextmanager:
+            self.cb_client.get_entity("roomDoesnotExist")
+        response = contextmanager.exception.response
+        self.assertEqual(response.status_code, 400)
+        self.assertEqual(response.json()["detail"], "Not a URL nor a URN")
+    
+    # TODO: write test which tries to delete entity with id AND type
+    # for orion-ld version 1.4.0, error BadRequestData (title: Unsupported URI parameter) happens
+    # def test_delete_entity_with_type(self):
+
+    def test_delete_entity(self):
+        """
+        Removes an specific Entity from an NGSI-LD system.
+        Args:
+            - entityID(string): Entity ID; required
+            - type(string): Entity Type
+        Returns: 
+            - (204) No Content. The entity was removed successfully.
+            - (400) Bad request.
+            - (404) Not found.
+        Tests:
+            - Try to delete an non existent entity -> Does it return a Not found?
+            - Post an entity and try to delete the entity -> Does it return 204?
+            - Try to get to delete an deleted entity -> Does it return 404?
+        """
+
+        """
+        Test 1:
+            delete entity with non existent entity_ID
+            If return != 404:
+                Raise Error
+        
+        Test 2: 
+            post an entity with entity_ID and entity_name
+            delete entity with entity_ID
+            get entity list 
+            If entity with entity_ID in entity list:
+                Raise Error
+        
+        Test 3: 
+            delete entity with entity_ID 
+                return != 404 ? 
+                    yes: 
+                        Raise Error
+        """
+            
+        """Test1"""
+        # try to delete nonexistent entity
+        with self.assertRaises(requests.exceptions.HTTPError) as contextmanager:
+            self.cb_client.get_entity(entity_id=self.entity.id)
+        response = contextmanager.exception.response
+        self.assertEqual(response.status_code, 404)
+        self.assertEqual(response.json()["title"], "Entity Not Found")
+
+        """Test2"""
+        self.cb_client.post_entity(entity=self.entity)
+        self.cb_client.post_entity(entity=self.entity_2)
+        entity_list = self.cb_client.get_entity_list()
+        self.assertEqual(len(entity_list), 2)
+        self.assertEqual(entity_list[0].id, self.entity.id)
```

Comment:
We can actually not ensure that the order of the entities is not changing.

Reply:
fixed